app/javascript/controllers/apex_controller.js

```diff
@@ -0,0 +1,90 @@
+import { Controller } from "@hotwired/stimulus"
+import ApexCharts from "apexcharts"
+
+const DEFAULT_OPTIONS = {
+  chart: {
+    height: "100%",
+    width: "100%",
+    zoom: { enabled: false },
+    animations: {
+      speed: 1000,
+      animateGradually: {
+        delay: 50,
+      },
+    },
+  },
+  tooltip: {
+    fixed: {
+      enabled: true,
+      position: "topRight",
+    },
+  },
+  yaxis: {
+    title: { text: "Downloads" },
+  },
+}
+
+const LINE_DEFAULTS = {
+  xaxis: { type: "datetime" },
+  stroke: {
+    curve: "smooth",
+    width: 2,
+  },
+  colors: [
+    "#007EB2",
+    "#FF9600",
+    "#75BBE1",
+    "#FFC107",
+    "#6F42C1",
+    "#DC3545",
+    "#198754",
+    "#D63384",
+    "#20C997",
+    "#555555",
+  ],
+  legend: {
+    show: false,
+  },
+}
+
+const BAR_DEFAULTS = {
+  plotOptions: {
+    bar: {
+      horizontal: true,
+    },
+  },
+}
+
+export default class extends Controller {
+  static values = {
+    id: String,
+    type: String,
+    series: Array,
+  }
+  static targets = ["chart", "episode"]
+
+  connect() {
```

Comment:
this gave me a little bit of a headache to figure out, merging objects. but essentially, there are a set of default options, i merge in the data that is input (`this.seriesValue`), and i merge in another set of config options based on what `type` of chart is selected.